packages/react-async/documentation/recipes.md

```diff
@@ -157,33 +147,124 @@ First, we create the async component for our order list. This component will pre
+    };
+  },
+});
+```
+
+Finally, follow the [`@shopify/react-async` route prefetching guide](https://github.com/Shopify/quilt/tree/master/packages/react-async#prefetchroute-and-prefetcher) to register this asynchronous component for a URL, or render the `OrderList.Prefetch` component manually before you expect the user to navigate to a page where this component is rendered, or use `usePrefetch(OrderList)` to get a function that will preload the scripts and data on demand.
+
+## UI that is initially hidden
+
+A significant amount of UI is often hidden until interacted with. This is particularly common for modals and popovers. These elements of the UI are often complex, and can sometimes have dedicated GraphQL queries. This makes them an ideal candidate for deferring and prefetching.
+
+In order to solve this, we can turn the modal or popover into an async component. We can then stop that component from loading until it is about to open (driven by an `open` prop, like the Polaris `Modal` component). Finally, we’ll make the component prefetch its GraphQL query, and we will force the component to prefetch when an action that shows the modal is hovered.
+
+Assuming the GraphQL query for a component has been constructed with `createAsyncQueryComponent` (as demonstrated in the example above), we can make the modal component async as follows:
+
+```tsx
+// in /app/sections/Orders/OdersList/components/CustomerOrdersModal/CustomerOrdersModal/index.tsx
+
+import {
+  createAsyncComponent,
+  usePreload,
+  usePrefetch,
+  useKeepFresh,
+} from '@shopify/react-async';
+import {SkeletonIndex} from 'components';
+import {CustomerOrdersModalQuery} from './components';
+
+export default createAsyncComponent({
+  load: () => import('./CustomerModal'),
+  defer: ({open}) => open,
+  usePreload: () => usePreload(CustomerOrdersModalQuery),
+  usePrefetch: () => usePrefetch(CustomerOrdersModalQuery),
+  useKeepFresh: () => useKeepFresh(CustomerOrdersModalQuery),
+});
+```
+
+The `defer` option can accept a function, which should return `true` when the component should begin to load. As noted in the previous example, this has a problem: there will be a delay after the user tries to open a modal before it is available to render, and an even longer delay until its GraphQL query has resolved. To fix this, we will preload the scripts for the component on mount, and we will prefetch the GraphQL query when the user hovers or starts touching the action that triggers the modal.
+
+```tsx
+// in /app/sections/Orders/OrderList/OrderList.tsx
+
+import React from 'react';
+import {usePrefetch} from '@shopify/react-async';
+import {CustomerOrdersModal} from './components';
+
+export default function CustomerOrdersModal() {
+  const [showModal, setShowModal] = useState(false);
+  const prefetch = usePrefetch(CustomerOrdersModal);
+
+  return (
     <>
-      <OrderListEmptyStateQuery.KeepFresh />
-      <OrderListQuery.KeepFresh />
+      <CustomerOrdersModal.Preload />
```

Comment:
This will call `usePreload(CustomerOrdersModalQuery)`, right?

What does it mean to `preload` an async query component?  

`prefetch` will send the API call. `preload` will fetch the bundle for the query component? (what was generated by https://github.com/Shopify/graphql-tools-web/tree/master/packages/graphql-mini-transforms)

Reply:
Yes, that's correct. Preload just brings in the assets, Prefetch runs the query, KeepFresh runs the query and polls.